src/Microsoft.AspNetCore.Build/NuGetInstall.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+
+namespace Microsoft.AspNetCore.Build
+{
+    public class NuGetInstall : Task
+    {
+        [Required]
+        public ITaskItem[] Packages { get; set; }
+
+        [Required]
+        public string DestinationFolder { get; set; }
+
+        public override bool Execute()
+        {
+            if (!Directory.Exists(DestinationFolder))
+            {
+                Directory.CreateDirectory(DestinationFolder);
+                Log.LogMessage($"Created destination folder: {DestinationFolder}");
+            }
+
+            var success = true;
+            var logger = new MSBuildNuGetLogger(Log);
+            foreach (var item in Packages)
+            {
+                // Load the package
+                var path = item.GetMetadata("FullPath");
+                if (!File.Exists(path))
+                {
+                    Log.LogError($"Package file not found: {path}");
+                    success = false;
+                }
+                else
+                {
+                    using (var stream = File.OpenRead(path))
+                    {
+                        PackageIdentity id;
+                        using (var reader = new PackageArchiveReader(stream, leaveStreamOpen: true))
```

Comment:
It may be... I do think we should always be referencing a pretty bleeding-edge NuGet to help avoid that. If we end up having to use our own code rather than the APIs we could do that too.